core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -726,12 +726,12 @@ package object config {
   private[spark] val SHUFFLE_SERVICE_DB_BACKEND =
     ConfigBuilder(Constants.SHUFFLE_SERVICE_DB_BACKEND)
       .doc("Specifies a disk-based store used in shuffle service local db. " +
-        "LEVELDB or ROCKSDB.")
+        "Only supported ROCKSDB at now.")
       .version("3.4.0")
       .stringConf
       .transform(_.toUpperCase(Locale.ROOT))
       .checkValues(DBBackend.values.map(_.toString).toSet)
-      .createWithDefault(DBBackend.LEVELDB.name)
+      .createWithDefault(DBBackend.ROCKSDB.name)
```

Comment:
I'm not against this change, but I'm quite conflicted about it.

For shuffle db, we didn't make `ROCKSDB` the default in Spark 3.5.0, so I'm not sure if it's appropriate to just remove it in Spark 4.0. This is why I haven't submitted a pr to address this issue.

Let's wait for other people's opinions.

Reply:
Agree, let us change to rocksdb as default first.
We can always remove it in 4.1 or later